### PR TITLE
move type and version to NLPacket headers

### PR DESCRIPTION
**Pull request by birarda**

### CODE REVIEW

File: libraries/octree/src/Octree.cpp [hunk 1]
```diff
@@ -1902,7 +1902,7 @@ bool Octree::readSVOFromStream(unsigned long streamLength, QDataStream& inputStr
     if (getWantSVOfileVersions()) {
 
         // read just enough of the file to parse the header...
-        const unsigned long HEADER_LENGTH = sizeof(PacketType::Value) + sizeof(PacketVersion);
+        const unsigned long HEADER_LENGTH = sizeof(PacketType) + sizeof(PacketVersion);
```
Comment (birarda) [inline libraries/octree/src/Octree.cpp]: need to make this so that it reads/writes the old size so we don't have problems loading old files

Comment (Atlante45) [inline libraries/octree/src/Octree.cpp]: Should be the same size though no?

Comment (birarda) [inline libraries/octree/src/Octree.cpp]: nah, the PacketType is a statically typed enum to uint8_t now

File: assignment-client/src/avatars/AvatarMixer.cpp [hunk 2]
```diff
@@ -259,8 +259,8 @@ void AvatarMixer::broadcastAvatarData() {
                         return;
                     }
 
-                    PacketSequenceNumber lastSeqToReceiver = nodeData->getLastBroadcastSequenceNumber(otherNode->getUUID());
-                    PacketSequenceNumber lastSeqFromSender = otherNode->getLastSequenceNumberForPacketType(PacketType::AvatarData);
+                    uint16_t lastSeqToReceiver = nodeData->getLastBroadcastSequenceNumber(otherNode->getUUID());
```
Comment (huffman) [inline assignment-client/src/avatars/AvatarMixer.cpp:262]: Why did we get rid of this typedef?

Comment (birarda) [inline assignment-client/src/avatars/AvatarMixer.cpp:262]: It no longer applies to all nodes and node types - I moved the sequence numbering we were doing for Avatars to be specific to Avatars. Could add a new type def if you think it'll be cleaner.

**Pull request merged**